lib/parameter_parser/email_alert_parameter_parser.rb

```diff
@@ -0,0 +1,98 @@
+# Ensures that email alert filter keys and values provided by users
+# and that Finder Frontend sends to Email Alert API are within the
+# set of permitted keys and values provided in the content item or
+# registries of people, organisations, taxons, and world locations.
+module ParameterParser
```

Comment:
A bit of a design question: do we want this to be *invalid* if there's an unrecognised filter given?  That would send the user back to the signup page with the message that something's wrong, rather than sign them up for something which doesn't match what they entered (even if they did somehow enter a bad facet)

Reply:
Same question for unrecognised values

Reply:
Yeah the intent is to remove any invalid filter keys and values and continue to sign users up. What solution do you prefer?

My initial implementation raised an error if there were any invalid keys and returned users to the signup page. I think that'd be a worse user experience because users who have bad query params won't be able to fix them on the signup page. The user would see their final filters in the title of their subscriber list, so they have a final chance to go back and fix their subscription.

I think ideally we'd display facets on email signup pages, so users could modify their subscription. It may not be obvious to users that their filters from the finder will be used in their subscription, e.g. when signing up to [News from HMRC](https://www.gov.uk/search/news-and-communications/email-signup?organisations%5B%5D=hm-revenue-customs) it's not obvious that they'll only get news from HMRC.

Reply:
Ah, I'd forgotten the list title includes what you're actually signing up for.  That's good, my concern was users being linked to a page which has a URL suggesting one thing, but they actually get signed up for a different thing.